lib/ts/recipe/emailpassword/components/library/formBase.tsx

```diff
@@ -251,10 +251,21 @@ export const FormBase: React.FC<FormBaseProps<any>> = (props) => {
                     // If field error.
                     if (result.status === "FIELD_ERROR") {
                         const errorFields = result.formFields;
-
-                        setFieldStates((os) =>
-                            os.map((fs) => ({ ...fs, error: errorFields.find((ef: any) => ef.id === fs.id)?.error }))
-                        );
+                        const getErrorMessage = (fs: FieldState) => {
+                            const errorMessage = errorFields.find((ef: any) => ef.id === fs.id)?.error;
+                            if (errorMessage === "Field is not optional") {
+                                const fieldConfigData = props.formFields.find((f) => f.id === fs.id);
+                                // replace non-optional server error message from nonOptionalErrorMsg
+                                if (
+                                    fieldConfigData?.nonOptionalErrorMsg !== undefined &&
+                                    fieldConfigData?.nonOptionalErrorMsg !== ""
+                                ) {
```

Comment:
Why are you checking for `!== ""`?

Reply:
What happens if the user sets it to `""`? Does it not render the optional error at all?

Reply:
the user will get the standard usual error  `Field is not optional` 
if the value of nonOptionalErrorMsg is set to empty string

Reply:
Yea, but why are we doing that?

Reply:
if we don't handle that 
- The end-user won't be able to submit the form & won't realize because of empty string

If we handle it
- It conveys the missing data in a usual way.